// react-img-mapper/dist/helpers/events.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.touchEnd = exports.touchStart = exports.mouseUp = exports.mouseDown = exports.mouseMove = exports.click = exports.mouseLeave = exports.mouseEnter = exports.imageClick = exports.imageMouseMove = void 0;
var imageMouseMove = function (props) { return function (event) {
    var onImageMouseMove = props.onImageMouseMove;
    if (onImageMouseMove)
        onImageMouseMove(event);
}; };
exports.imageMouseMove = imageMouseMove;
var imageClick = function (props) { return function (event) {
    var onImageClick = props.onImageClick;
    if (onImageClick) {
        event.preventDefault();
        onImageClick(event);
    }
}; };
exports.imageClick = imageClick;
var mouseEnter = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onMouseEnter = props.onMouseEnter, cb = props.cb;
        if (cb)
            cb(area);
        if (onMouseEnter)
            onMouseEnter(area, index, event);
    };
};
exports.mouseEnter = mouseEnter;
var mouseLeave = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onMouseLeave = props.onMouseLeave, cb = props.cb;
        if (cb)
            cb(area);
        if (onMouseLeave)
            onMouseLeave(area, index, event);
    };
};
exports.mouseLeave = mouseLeave;
var click = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onClick = props.onClick, cb = props.cb;
        if (cb)
            cb(area);
        if (onClick) {
            event.preventDefault();
            onClick(area, index, event);
        }
    };
};
exports.click = click;
var mouseMove = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onMouseMove = props.onMouseMove;
        if (onMouseMove)
            onMouseMove(area, index, event);
    };
};
exports.mouseMove = mouseMove;
var mouseDown = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onMouseDown = props.onMouseDown;
        if (onMouseDown)
            onMouseDown(area, index, event);
    };
};
exports.mouseDown = mouseDown;
var mouseUp = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onMouseUp = props.onMouseUp;
        if (onMouseUp)
            onMouseUp(area, index, event);
    };
};
exports.mouseUp = mouseUp;
var touchStart = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onTouchStart = props.onTouchStart;
        if (onTouchStart)
            onTouchStart(area, index, event);
    };
};
exports.touchStart = touchStart;
var touchEnd = function (_a, props) {
    var area = _a.area, index = _a.index;
    return function (event) {
        var onTouchEnd = props.onTouchEnd;
        if (onTouchEnd)
            onTouchEnd(area, index, event);
    };
};
exports.touchEnd = touchEnd;
